contracts/sol6/KyberNetwork.sol

```diff
@@ -1184,16 +1241,20 @@ contract KyberNetwork is WithdrawableNoModifiers, Utils5, IKyberNetwork, Reentra
                     totalWeightedDestAmount;
             destAmountSoFar += destAmountSplit;
 
-            uint256 newSrcAmount = calcSrcQty(
+            newSrcAmounts[i] = calcSrcQty(
                 destAmountSplit,
                 srcDecimals,
                 destDecimals,
                 reservesData.rates[i]
             );
-            newSrcAmount = minOf(newSrcAmount, currentSrcAmount);
+            if (newSrcAmounts[i] > currentSrcAmount) {
+                // revert back to use current src amounts
+                newSrcAmount = srcAmount;
+                return newSrcAmount;
+            }
 
-            reservesData.srcAmounts[i] = newSrcAmount;
-            srcAmount += newSrcAmount;
+            newSrcAmount += newSrcAmounts[i];
         }
+        reservesData.srcAmounts = newSrcAmounts;
```

Comment:
maybe add comment. 
// new src amounts used only any of them isn't greater then current srcAmount.